src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -1,1219 +1,1227 @@
+                {
+                    //create a string collection of the assigned letters
+                    var groupPermissionCodes = groupPermissions.ToArray();
+
+                    //check if there are no permissions assigned for this group save model, if that is the case we want to reset the permissions
+                    //for this group/node which will go back to the defaults
+                    if (groupPermissionCodes.Length == 0)
+                    {
+                        Services.UserService.RemoveUserGroupPermissions(userGroup.Id, content.Id);
+                    }
+                    //check if they are the defaults, if so we should just remove them if they exist since it's more overhead having them stored
+                    else if (userGroup.Permissions.UnsortedSequenceEqual(groupPermissionCodes))
+                    {
+                        //only remove them if they are actually currently assigned
+                        if (contentPermissions.ContainsKey(userGroup.Id))
+                        {
+                            //remove these permissions from this node for this group since the ones being assigned are the same as the defaults
+                            Services.UserService.RemoveUserGroupPermissions(userGroup.Id, content.Id);
+                        }
+                    }
+                    //if they are different we need to update, otherwise there's nothing to update
+                    else if (contentPermissions.ContainsKey(userGroup.Id) == false || contentPermissions[userGroup.Id].AssignedPermissions.UnsortedSequenceEqual(groupPermissionCodes) == false)
+                    {
+
+                        Services.UserService.ReplaceUserGroupPermissions(userGroup.Id, groupPermissionCodes.Select(x => x[0]), content.Id);
+                    }
+                }
+            }
+
+            return GetDetailedPermissions(content, allUserGroups);
+        }
+
+        /// <summary>
+        /// Returns the user group permissions for user groups assigned to this node
+        /// </summary>
+        /// <param name="contentId"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// Permission check is done for letter 'R' which is for <see cref="ActionRights"/> which the user must have access to to view
+        /// </remarks>
+        [EnsureUserPermissionForContent("contentId", 'R')]
+        public IEnumerable<AssignedUserGroupPermissions> GetDetailedPermissions(int contentId)
+        {
+            if (contentId <= 0) throw new HttpResponseException(Request.CreateResponse(HttpStatusCode.NotFound));
+            var content = Services.ContentService.GetById(contentId);
+            if (content == null) throw new HttpResponseException(Request.CreateResponse(HttpStatusCode.NotFound));
+
+            //TODO: Should non-admins be able to see detailed permissions?
+
+            var allUserGroups = Services.UserService.GetAllUserGroups();
+
+     
```

Comment:
Shouldn't need to modify content controller at all, all changes will come from the EntityController